### PR TITLE
[WIP] Add Jenkinsfile, facelift and fix build

### PR DESCRIPTION
**Pull request by batmat**

### CODE REVIEW

**Comment by batmat** [comment]
Waiting on https://issues.jenkins-ci.org/browse/INFRA-2137 before pushing more commits

**Comment by oleg-nenashev** [comment]
Not sure it will work for a non-plugin component.
See https://github.com/jenkins-infra/pipeline-library/pull/56

Maybe a copy-paste of https://github.com/jenkinsci/custom-war-packager/blob/master/Jenkinsfile would do the job for now

**Comment by batmat** [comment]
ok, build seems to work, but failing because there's no test and Surefire is configured to bark in this case. I think I'll just add a first dummy test to get past this.

File: plugin-management-cli/pom.xml [hunk 1]
```diff
@@ -3,13 +3,20 @@
          xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
          xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
     <parent>
-        <artifactId>plugin-manager</artifactId>
-        <groupId>jenkins</groupId>
+        <groupId>io.jenkins.plugin-management</groupId>
```
Comment (kwhetstone) [inline plugin-management-cli/pom.xml]: +1 and for PR3, the package names can renamed during that cleanup

File: pom.xml [hunk 2]
```diff
@@ -4,14 +4,21 @@
...
-    <groupId>jenkins</groupId>
-    <artifactId>plugin-management</artifactId>
+    <parent>
+        <groupId>org.jenkins-ci</groupId>
+        <artifactId>jenkins</artifactId>
+        <version>1.51</version>
+    </parent>
     <packaging>pom</packaging>
+    <groupId>io.jenkins.plugin-management</groupId>
+    <artifactId>jenkins-plugin-management-parent-pom</artifactId>
```
Comment (kwhetstone) [inline pom.xml]: Maybe the new artifact id should match the repository name `jenkins-plugin-installation-manager`

Comment (stopalopa) [reply pom.xml]: I already merged this. I also thought that the artifactId had to match the repository name, but for the custom war packager it also does not match: https://github.com/jenkinsci/custom-war-packager/blob/master/pom.xml

File: Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1 @@
+buildPlugin(configurations: buildPlugin.recommendedConfigurations())
```
[the remark this answers is not in the document]
Comment (stopalopa) [reply Jenkinsfile]: Sorry, I was fixing the merge conflicts and ended up merging everything. If this is going to be a problem, I can take it out.

**Comment by oleg-nenashev** [comment]
As I said, the `buildPlugin()` pipeline does not work for non-plugin components anymore: https://github.com/jenkinsci/plugin-installation-manager-tool/pull/2#issuecomment-499536248

```
[2019-06-08T12:36:36.080Z] Archiving artifacts
[2019-06-08T12:36:36.101Z] ‘**/target/*.hpi’ doesn’t match anything: ‘**’ exists but not ‘**/target/*.hpi’
No artifacts found that match the file pattern "**/target/*.hpi,**/target/*.jpi". Configuration error?
```

**Comment by kwhetstone** [comment]
PR#3 should fix the pipeline; we know it's failing and have an idea of what needs to be added to build successfully.

**Pull request merged**